classes/services/PKPSubmissionService.inc.php

```diff
@@ -747,11 +747,36 @@ public function delete($submission) {
 	 */
 	public function canUserEditMetadata($submissionId, $userId) {
 		$stageAssignments = DAORegistry::getDAO('StageAssignmentDAO')->getBySubmissionAndUserIdAndStageId($submissionId, $userId, null);
+		// Check for permission from stage assignments
 		while ($stageAssignment = $stageAssignments->next()) {
 			if ($stageAssignment->getCanChangeMetadata()) {
 				return true;
 			}
 		}
+		// If user has no stage assigments, check if user can edit anyway ie. is manager
+		$context = Application::get()->getRequest()->getContext();
+		if ($this->_canUserAccessUnassignedSubmissions($context->getId(), $userId)){
+			return true;
+		}
```

Comment:
I may be reading this wrong, but I think this will allow a `ROLE_ID_MANAGER` to edit metadata even if they have a different stage assignment. I think the conditional needs to be like:

```
if ($stageAssignments->wasEmpty() && $this->_canUserAccessUnassignedSubmissions(...)) {
```

Reply:
true and check, but hey the correct logic was at least in the comment above :)